numba/core/interpreter.py

```diff
@@ -569,6 +583,12 @@ def _remove_unused_temporaries(self):
                     inst.value.value = replaced_var[inst.value.value.name]
                     new_body.append(inst)
                     continue
+
+                # iterate over RHS operands and replace them iff they
+                # are in replaced_var dict
+                if len(replaced_var) and isinstance(inst.value, ir.Expr):
```

Comment:
May main concern is that this doesn't handle all nodes and may not be robust. I think we can probably come up with a more robust solution for the byte code pattern of walrus operator.

This may also slow down compilation. Has there been any compile time benchmarking for this?